mlflow/utils/model_utils.py

```diff
@@ -74,8 +75,8 @@ def _get_flavor_configuration_from_uri(model_uri, flavor_name, logger):
                 f"Falling back to downloading from original model URI {model_uri}",
                 exc_info=True,
             )
-            ml_model_file = _download_artifact_from_uri(
-                artifact_uri=append_to_uri_path(model_uri, MLMODEL_FILE_NAME)
+            ml_model_file = get_artifact_repository(artifact_uri=model_uri).download_artifacts(
```

Comment:
Instead of assuming that we can append `MLmodel` to an artifact URI to get a valid artifact URI (which is not generally true), we instead construct an `ArtifactRepository` against the original artifact URI (e.g. `models:/mymodels@alias`) and then call `download_artifacts` with the actual file path (`MLModel`). We should make similar fixes elsewhere, I'll do that in a follow-up PR

Reply:
Hm, not sure my suggestion will work. The problem is that we don't restrict the set of allowed characters in registered model names in OSS. That means that a URI like `models:/mymodel@myalias/1/to/file` could ambiguously be interpreted as (1) targeting the file at path `1/to/file` under the version of `mymodel` associated with the alias `myalias` or (2) targeting the model named `mymodel@alias`, version `1`, file path `/to/file`.